src/map/unit.c

```diff
@@ -827,13 +827,7 @@ static bool unit_run(struct block_list *bl, struct map_session_data *sd, enum sc
 		to_y += dir_y;
 	}
 
-	// Can't run forward
-	if( (to_x == bl->x && to_y == bl->y ) || (to_x == (bl->x+1) || to_y == (bl->y+1)) || (to_x == (bl->x-1) || to_y == (bl->y-1))) {
```

Comment:
No. What case do you think is different? I was pretty sure it was going to be equivalent.

Reply:
i not see in other functions same check. may be similar but not same checks present,
probably better left fast fail as is here

Reply:
@4144 

Explaining this based on the following commit: https://github.com/skyleo/Hercules/commit/0964a8869a361f24800fec664591eb43bc252d83
Alright so:
**First:**
If `to_x == bl->x && to_y == bl->y` then`i == 0`
so we don't return on `if (i > 1 ...)`

*)
```
	do {
		to_x -= dir_x;
		to_y -= dir_y;
	} while (--i > 0 && unit->walk_toxy(bl, to_x, to_y, 1) != 0);
```
is executed but to_x and to_y are just local variables so we don't care about any changes since `--i > 0` is false in the case of `i == 0`
so
```
	if (i <= 0) {
		unit->run_hit(bl, sc, sd, type);
		return false;
	}
```
is executed fully, which is the old behavior, same order of execution with 2 unnecessary subtractions but that's neglible.

**Other case**
if `(to_x == (bl->x+1) || to_y == (bl->y+1)) || (to_x == (bl->x-1) || to_y == (bl->y-1))`
which is equivalent to `to_x == bl->x + 1 || to_y == bl->y + 1 || to_x == bl->x - 1 || to_y == bl->y - 1`
which then means that since dir_x and dir_y are in {-1, 1, 0}
must mean that to_x and to_y (which initially are set to bl->x and bl->y)
have had exactly 1 addition of dir_x and dir_y
which then means that `i == 1`

so `i > 1` is false and the rest is analog to *)

So as we can see the logic did not change. *q.e.d* :)

I will only not do this if someone can disprove me with a counter-example.